pkg/compose/convergence.go

```diff
@@ -291,10 +290,7 @@ func (s *composeService) waitDependencies(ctx context.Context, project *types.Pr
 			continue
 		}
 
-		containers, err := s.getContainers(ctx, project.Name, oneOffExclude, false, dep)
-		if err != nil {
-			return err
-		}
+		containers = containers.filter(isService(dep))
```

Comment:
Shouldn't we use a dedicated variable to manage filtered containers? I wondering if we won't loose potential dependencies by re-assigning `containers`

Reply:
indeed, nice catch - I've been lazy here